src/Http/Routing/src/Matching/CandidateSet.cs

```diff
@@ -244,15 +124,7 @@ public bool IsValidCandidate(int index)
                 ThrowIndexArgumentOutOfRangeException();
             }
 
-            if (Count < BitVectorSize)
-            {
-                // Get the n-th bit
-                return _validity[0x00000001 << index];
-            }
-            else
-            {
-                return _largeCapactityValidity[index];
-            }
+            return _candidates[index].Score >= 0;
```

Comment:
We're able to simplify this dramatically by using the score to store whether or not the candidate is valid